### PR TITLE
Changes for Java9+

### PR DESCRIPTION
**Pull request by georgweiss**
Tweaks to pom file due to changed behaviour of the javadoc tool. A change in java source and target version is also needed to compile successfully.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,6 @@
             <name>OSS Snapshots</name>
             <url>https://oss.sonatype.org/content/repositories/snapshots</url>
         </repository>
-        <!-- Temporary - needed for old snapshots of ChannelfinderAPI -->
```
Comment (ralphlange) [inline pom.xml:28]: Why remove this comment? It still applies.

Comment (georgweiss) [reply pom.xml:28]: Well, if it still applies then it's not temporary?

Comment (ralphlange) [reply pom.xml:28]: Let me explain.
The ChannelFinder API has changed. The only place where Maven can find the old versions is the deprecated EPICS repository on SourceForge.
As soon as the directoryService example has been updated to use the current API of ChannelFinder, the pointer to this deprecated repository should be removed.

Is that "temporary"? Do you have a better term?

Comment (georgweiss) [reply pom.xml:28]: Then I think the comment should state what you just wrote. Plus the version numbers for the "old" and "current" ChannelFinder API.

**Comment by ralphlange** [comment]
In general, I prefer unrelated changes being in separate commits, but as in this case this is really minor, I will merge anyway.

**Comment by georgweiss** [comment]
Thanks.

**Pull request merged**